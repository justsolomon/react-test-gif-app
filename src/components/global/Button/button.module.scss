@import "../../../assets/scss/base/variables";

.button {
  height: 40px;
  padding: 0 1rem;
  border: none;
  cursor: pointer;
  color: #fff;
  font-size: $font-size--md;
  font-weight: $font-weight--bold;
  border-radius: $border-radius--sm;

  &:disabled {
    cursor: not-allowed;
    opacity: 0.7;
  }

  &--solid,
  &--primary {
    background-color: $brand-color--primary;

    &:hover {
      background-color: $brand-color--secondary;
    }

    &:active {
      background-color: $brand-color--tertiary;
    }

    &:disabled:hover {
      background-color: $brand-color--primary;
    }
  }

  &--secondary {
    background-color: $bg-light--primary;
    color: $primary-color;

    &:hover {
      background-color: $bg-light--secondary;
    }

    &:active {
      background-color: $bg-light--tertiary;
    }

    &:disabled:hover {
      background-color: $bg-light--primary;
    }
  }

  &--ghost,
  &--transparent {
    background-color: transparent;

    &:disabled:hover {
      background-color: transparent;
    }
  }

  &--ghost {
    &:hover {
      background-color: $bg-light--primary;
    }

    &:active {
      background-color: $bg-light--secondary;
    }
  }

  &--full {
    width: 100%;
  }
}
