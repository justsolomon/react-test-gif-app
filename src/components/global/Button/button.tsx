import styles from "./button.module.scss";

interface IButton {
  children: React.ReactNode;
  type?: "button" | "reset" | "submit";
  className?: string;
  isLoading?: boolean;
  isDisabled?: boolean;
  isFullWidth?: boolean;
  variant?: "solid" | "ghost" | "transparent";
  colorScheme?: "primary" | "secondary" | "none";
  style?: object;
  onClick?: React.MouseEventHandler;
}

function Button({
  isDisabled,
  isLoading,
  children,
  className = "",
  isFullWidth,
  style,
  type = "button",
  variant = "solid",
  colorScheme = "none",
  onClick
}: IButton) {
  return (
    <button
      type={type}
      className={`${styles["button"]} ${className} ${
        styles["button--" + variant]
      } ${styles["button--" + colorScheme]} ${
        isFullWidth ? styles["button--full"] : ""
      }`}
      disabled={isDisabled || isLoading}
      style={style}
      onClick={onClick}
    >
      {children}
    </button>
  );
}

export default Button;
