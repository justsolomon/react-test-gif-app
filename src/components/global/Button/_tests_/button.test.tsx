import { render } from "@testing-library/react";
import Button from "../button";

describe("<Button />", () => {
  it("should render the button component properly", () => {
    const { asFragment } = render(<Button>Test button</Button>);

    expect(asFragment()).toMatchSnapshot();
  });
});
