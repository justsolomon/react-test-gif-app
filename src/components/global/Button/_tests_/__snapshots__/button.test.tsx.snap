// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button component properly 1`] = `
<DocumentFragment>
  <button
    class="button  button--solid button--none "
    type="button"
  >
    Test button
  </button>
</DocumentFragment>
`;
